articles/an-introduction-to-graph-neural-network/index.md

```diff
@@ -0,0 +1,188 @@
+This has led to an increasingly growing interest in deep learning research that focuses on the structure of graph data.
+
+In this article, we provide a comprehensive overview of:
+
+1. What is a Graph?
+
+2. What is Graph Neural Networks?
+
+3. Graph Learning Python Libraries
+
+4. Models of Graph Neural Networks
+
+5. Applications of Graph Neural Networks
+
+
+### Prerequisites
+
+This article assumes a basic understanding of Machine Learning (ML) and Deep Learning (DL). For an introduction to ML and DL, feel free to check out my previous [article](/https://www.section.io/engineering-education/differences-between-artificial-intelligence-machine-learning-and-deep-learning//) 
+
+### What is a Graph?
+
+In Computer Science, Graphs are data structures that model a set of objects and their relationships. These objects represent the nodes and the relationships represent edges. 
+
+![A Graph data structure](/engineering-education/an-introduction-to-graph-neural-network/edges-links.png)
+
+Let's assume a graph, G. This graph can be described by:
+* V as the vertex set 
+* E as the edges
+
+Then,   G = (V,E)
+
+It is important to note that the vertex, V, often refers to the nodes.
+
+Let's take an example of a transportation system. This can be thought of as a graph. The locations in the system represent the vertice, V, and the connections between these locations representing the edges.
+
+### What is Graph Neural Network (GNN)?
+
+Graph Neural Networks is a technique in deep learning which extends existing neural networks for processing data on graphs. 
+
+GNNs motivation stems from three areas of study:
+
+* Convolution Neural Networks (CNN)
+These are deep neural networks used to analyze image data. It is used to solve image processing tasks.
+
+* Recurrent Neural Networks (RNN)
+RNN is a type of neural network that passes information internally from one time step to the next to process a sequence. It's commonly used to solve Natural Language Processing (NLP) and speech recognition tasks.  
+
+* Auto-Encoders (AE)
+This is a type of Artificial Neural Network (ANN). It consists of an encoder, code, and a decoder. It encodes the representations of a set of data. It then decodes the reduced encoded data generating new data similar to that of its original input. 
+
+Based on these three motivations, GNNs can collectively combine information from the graph structure.
+
+Common applications of GNN are in node classification, link prediction, and graph classification. 
+
+In **node classification**, every node in the network is assigned a label. The network can classify new nodes introduced without the ground truth.
+In **link prediction**, the goal is to  predicts the likelihood of two nodes being inter-linked.
+In **graph classification**, the idea is to classify graphs into different classes. This is related to image classification but the target changes into classifying graphs rather than images.
+
+### Graph Learning Python Libraries
```

Comment:
Why would one have to use libraries? What advantages are these libraries giving? Are they modular and efficient? Let the user know why such libraries are required. Yes, to implement GNNs. But, give some more additional information. 

It would be worth guiding the reader to a project implemented using either of these libraries and briefly explaining the project. That way, readers will know what to expect from the libraries and what to use these for.